src/main.rs

```diff
@@ -49,24 +49,58 @@ pub fn get_network(network_str: &str) -> &str {
     }
 }
 
+// At the top level, make the signal handler accessible throughout the program
+lazy_static::lazy_static! {
+    static ref SIGNAL_RECEIVED: Arc<AtomicBool> = Arc::new(AtomicBool::new(false));
+}
+
 #[tokio::main]
 async fn main() -> anyhow::Result<()> {
     // Handle SIGTERM and SIGINT gracefully by stopping the docker container
     let mut signals = Signals::new([SIGTERM, SIGINT])?;
     let mut container_id: Option<String> = None;
     let mut temp_dir: Option<String> = None;
-    let caught_signal = Arc::new(AtomicBool::new(false));
 
-    let caught_signal_clone = caught_signal.clone();
     let handle = signals.handle();
     std::thread::spawn(move || {
-        #[allow(clippy::never_loop)]
-        for _ in signals.forever() {
-            caught_signal_clone.store(true, Ordering::Relaxed);
-            break;
+        if signals.forever().next().is_some() {
+            SIGNAL_RECEIVED.store(true, Ordering::Relaxed);
         }
     });
 
+    // Add a function to check if we should abort
+    let check_signal = |container_id: &mut Option<String>, temp_dir: &mut Option<String>| {
+        if SIGNAL_RECEIVED.load(Ordering::Relaxed) {
+            println!("\nReceived interrupt signal, cleaning up...");
+
+            if let Some(container_id) = container_id.take() {
+                if std::process::Command::new("docker")
+                    .args(["kill", &container_id])
+                    .output()
+                    .is_err()
+                {
+                    println!("Failed to close docker container");
+                } else {
+                    println!("Stopped container {}", container_id)
+                }
+            }
+
+            if let Some(temp_dir) = temp_dir.take() {
+                if std::process::Command::new("rm")
+                    .args(["-rf", &temp_dir])
+                    .output()
+                    .is_err()
+                {
+                    println!("Failed to remove temporary directory");
+                } else {
+                    println!("Removed temporary directory {}", temp_dir);
+                }
+            }
+
+            std::process::exit(130);
```

Comment:
why use this specific exit code?

Reply:
Its the exit code for ctrl+c https://tldp.org/LDP/abs/html/exitcodes.html

Reply:
Ill remove that an see of just breaking the loop is also enough. I had a hard time getting this endless API request to stop.